lib/timers.js

```diff
@@ -283,7 +283,10 @@ function listOnTimeout(list, now) {
     // Check if this loop iteration is too early for the next timer.
     // This happens if there are more timers scheduled for later in the list.
     if (diff < msecs) {
+      const prev = list.expiry;
       list.expiry = timer._idleStart + msecs;
+      if (list.expiry <= prev)
+        list.expiry++;
```

Comment:
Can you please add a comment here explaining this is done to avoid floating point rounding errors? It's clear in the context of this PR but less so when seen in isolation.

Reply:
Sure, but it is a little difficult to explain just a few lines for me.. I would like you to help it.

https://github.com/nodejs/node/blob/af6d26281f734dbbf1bf497103d1cb4b85a52b2e/lib/timers.js#L206-L207

 ```js
const msecs = 1.00000000000001;
console.log(126 + msecs); // 127.00000000000001
console.log(127 + msecs); // 128
```
In this case, when `start` is turning to `127`, floating point precison is changed. So it will be integer.

Then,
https://github.com/nodejs/node/blob/af6d26281f734dbbf1bf497103d1cb4b85a52b2e/lib/timers.js#L259-L263

when `now` is `128`, it will be `false`.

After that, `listOnTimeout` will be called and check the differences here.


https://github.com/nodejs/node/blob/af6d26281f734dbbf1bf497103d1cb4b85a52b2e/lib/timers.js#L285-L288

In this time, the `diff` will be `1` but `msecs` is `1.00000000000001`. 
So the if state will be `true` and `queue.percolateDown(1)` will be called. 
Finally, the `list` will be peeked and goes into an infinite loop .

I could change `list.expiry > now` to `list.expiry >= now`, but I'm afraid it may break something else.